sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3532,22 +3509,29 @@ case class ArrayDistinct(child: Expression)
              |}
            """.stripMargin
 
-        val processArray = withArrayNullAssignment(
-          s"$jt $value = ${genGetValue(array, i)};" +
-            SQLOpenHashSet.withNaNCheckCode(elementType, value, hashSet, body,
-              (valueNaN: String) =>
-                s"""
-                   |$size++;
-                   |$builder.$$plus$$eq($valueNaN);
-                   |""".stripMargin))
+        val processArray = SQLOpenHashSet.withNullCheckCode(dataType, dataType, hashSet,
```

Comment:
can we follow the code style of the interpreted code path?
```
val withNaNCheckCodeGenerator = (array: String, index: String) => ...
val processArray = SQLOpenHashSet.withNullCheckCode(..., array => withNaNCheckCodeGenerator(array, _), ...)
```

Reply:
Done